esmvaltool/cmorizers/data/formatters/datasets/esacci_seaice.py

```diff
@@ -0,0 +1,319 @@
+                  'inputfilelist', 'history', 'valid_min', 'valid_max']
+
+    new_list = iris.cube.CubeList()
+
+    for cube in cube_list:
+        for attr in drop_attrs:
+            if attr in cube.attributes.keys():
+                cube.attributes.pop(attr)
+
+        new_list.append(cube)
+
+    # make sure there is one cube for every day (daily data) or
+    # every month (monthly data) of the year
+    # (print debug info about missing days/months and add cube with
+    # nan to fill gaps
+
+    full_list = iris.cube.CubeList()
+    time_list = []
+
+    for cube in new_list:
+        loncoord = cube.coord('longitude')
+        latcoord = cube.coord('latitude')
+        loncoord.points = np.round(loncoord.core_points(), 3)
+        latcoord.points = np.round(latcoord.core_points(), 3)
+
+    # create list of available days/months ('time_list')
+
+    for cube in new_list:
+        timecoord = cube.coord('time')
+        cubetime = timecoord.units.num2date(timecoord.points)
+        ctnew = cubetime[0].replace(hour=0, minute=0, second=0, microsecond=0)
+        time_list.append(ctnew)
+
+    # create cube list for every day/month of the year by adding
+    # cubes containing only nan to fill possible gaps
+
+    if is_daily:
+        loop_date = datetime(year0, 1, 1)
+        while loop_date <= datetime(year0, 12, 31):
+            date_available = False
+            for idx, cubetime in enumerate(time_list):
+                if loop_date == cubetime:
+                    date_available = True
+                    break
+            if date_available:
+                full_list.append(new_list[idx])
+            else:
+                logger.debug("No data available for %d/%d/%d", loop_date.month,
+                             loop_date.day, loop_date.year)
+                nan_cube = _create_nan_cube(new_list[0], loop_date.year,
+                                            loop_date.month, loop_date.day,
+                                            is_daily)
+                full_list.append(nan_cube)
+            loop_date += relativedelta.relativedelta(days=1)
+    else:
+        loop_date = datetime(year0, 1, 15)
+        while loop_date <= datetime(year0, 12, 31):
+            date_available = False
+            for idx, cubetime in enumerate(time_list):
+                if loop_date == cubetime:
+                    date_available = True
+                    break
+            if date_available:
+                full_list.append(new_list[idx])
+            else:
+                logger.debug("No data available for %d/%d", loop_date.month,
+                             loop_date.year)
+                nan_cube = _create_nan_cube(new_list[0], loop_date.year,
+                                            loop_date.month, loop_date.day,
+                                            is_daily)
+                full_list.append(nan_cube)
+            loop_date += relativedelta.relativedelta(months=1)
```

Comment:
Same here, data is daily by default, should we keep the code for monthly frequency? It is not used

Reply:
Unused code has been removed: https://github.com/ESMValGroup/ESMValTool/pull/3821/commits/c2e557aa29ef1b69cc711a967f2c450370a3a577